src/averageAges.js

```diff
@@ -14,12 +14,21 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+
+const oneHundredYears = 100;
+const maleSex = 'm';
+const femaleSex = 'f';
+
+function calculateMenAverageAge(people, century = null) {
+  const men = people.filter(person => person.sex === maleSex);
+  const filteredMen = century !== null
+    ? men.filter(person => Math.ceil(person.died
+      / oneHundredYears) === century)
+    : men;
+  const sumAge = filteredMen.reduce((sum, person) =>
+    sum + (person.died - person.born), 0);
+
+  return sumAge / filteredMen.length;
```

Comment:
You make a similar calculation for `women`, so it should be transferred to a separate function and re-used.